tls/s2n_config.c

```diff
@@ -554,8 +557,14 @@ int s2n_config_add_dhparams(struct s2n_config *config, const char *dhparams_pem)
     GUARD(s2n_alloc(&mem, sizeof(struct s2n_dh_params)));
     config->dhparams = (struct s2n_dh_params *)(void *)mem.data;
 
-    GUARD(s2n_stuffer_alloc_ro_from_string(&dhparams_in_stuffer, dhparams_pem));
-    GUARD(s2n_stuffer_growable_alloc(&dhparams_out_stuffer, strlen(dhparams_pem)));
+    if (s2n_stuffer_alloc_ro_from_string(&dhparams_in_stuffer, dhparams_pem)) {
+        s2n_free(&mem);
```

Comment:
Similar to the above, multiple `s2n_free` could justify a `DEFER_CLEANUP` for the blob.

Reply:
Again, I think this memory gets used outside of this function and would seg fault later if we freed up mem here.